artemis/src/main/java/tech/pegasys/artemis/BeaconNode.java

```diff
@@ -40,6 +48,19 @@
 
 public class BeaconNode {
   private static final ALogger LOG = new ALogger(BeaconNode.class.getName());
+  private final Vertx vertx = Vertx.vertx();
+  private final ExecutorService threadPool =
+      Executors.newCachedThreadPool(
+          new ThreadFactory() {
+            @Override
+            public Thread newThread(@NotNull Runnable r) {
+              Thread t = new Thread(r);
+              t.setDaemon(true);
```

Comment:
Why are we using daemon threads?

Reply:
We want the thread to stop when the JVM exits. I need to check the behavior of Guava's thread pool to make sure I am not keeping threads alive - it is on my mental list of things to do

Reply:
I think daemon threads are specifically built to stay alive after JVM exits. This is counter to your purpose.

Reply:
sorry that was wrong.  daemon threads will be killed after all other non daemon threads stop.    the point of using the daemon is that i just want this running in the background as low priority.